src/write_gcm.c

```diff
@@ -519,12 +516,16 @@ typedef struct {
   size_t size;
 } wg_curl_write_ctx_t;
 
-static size_t wg_curl_write_callback(char *ptr, size_t size, size_t nmemb,
+static int wg_curl_get_or_post(wg_curl_write_ctx_t *ctx, const char *url,
```

Comment:
I'd move this declaration back to its original spot and move the struct declaration above that.

Reply:
Roger that, updated.

Reply:
See my proposal that does not involve using this struct in the function's public API.